### PR TITLE
some work on the install script

### PR DESCRIPTION
**Pull request by amtoine**
This PR should address most of the points in #35.

### CODE REVIEW

**Comment by amtoine** [comment]
## changelog
> see `git diff 1e6ca0e^..ceeee65` or the [diff on _GitHub_](https://github.com/goatfiles/dotfiles/compare/1e6ca0e^..ceeee65)
>
here, i fixed most of the points in #35 and refactored the script into functions

> see `git diff 6bf4098^..e46bb21` or the [diff on _GitHub_](https://github.com/goatfiles/dotfiles/compare/6bf4098^..e46bb21)
>
here, i refactored all the hardcoded values into constants at the top and created lists of things to iterate over them and avoid redundant code, such as the `./install.sh x86_64/...` or `sudo systemctl enable ...`

> see `git diff 842a8da^..31a8bdc` or the [diff on _GitHub_](https://github.com/goatfiles/dotfiles/compare/842a8da^..31a8bdc)
>
in these last changes, i tried to fix bugs that occured in a VM, mainly typos and pulling the files at the right places for now :+1: 

## remaining bugs to be fixed and features to be implemented
- the names of the packages are not correct and thus fail the install, without stopping
should be addressed in #67 
- the `cargo` dependencies do not install properly
work is to be done in the `install_dependencies` function
- the remaining points in #35

**Comment by amtoine** [comment]
## changelog
> see `git diff f25476e^..b7fc147` or the [diff on _GitHub_](f25476e^..b7fc147)

#### regarding #35 itself
- i could not find the problematic files i talk about in the last comment of #35, i won't implement the feature
- thanks to the early quitting on any error i've implemented, i think a log file is not usefull here

#### the changes
in the order that they appear in the [f25476e^..b7fc147](f25476e^..b7fc147) above
- the dependencies are quite hard to install right now...
i've hardcoded the mandatory packages for the GOAT system to launch and disabled the installation from `pkgs.toml`.
see conclusion below
- system files from `.system/` are installed in `/`
- a new `echo_and_run` to run commands but see them clearly at the same time
  - this function `exit`s when the command given to it fails
  - subshell, i.e. for the `pkgbuilds` and `dmenu` take that error code and propagate the error
the result of this is that `install.sh` now quits on any error :+1: 

## conclusion
apart from the missing dependencies, i've tried the script in a VM and it runs until completion :ok_hand:
- could you try to run it as well @atxr? 

there sould be an error on the `paru` packages, as the name is not correct anymore :point_right: this is addressed in #67 and applying the same changes works fine as well :+1: 

after that, the script should work fine and print a :green_circle: message at the end :tada: 
you can reboot and have an incomplete system, but running :+1: 

things like `dmscripts`, `ghq`, ..., are missing
i would like to fix all of these dependency issues once and for all in #43, and leave the script as-is for now :relieved:

File: install.sh [hunk 1]
```diff
@@ -10,52 +10,182 @@
...
+
+REMOTE_DMENU="https://github.com/bakkeby/dmenu-flexipatch"
+LOCAL_DMENU_DIR="/tmp/dmenu"
+
+REMOTE_PKGBUILDS="https://github.com/goatfiles/pkgbuilds"
+LOCAL_PKGBUILDS_DIR="/tmp/pkgbuilds"
+
+PKGBUILDS=(
+    paru
+    amtoine-scripts-git
```
Comment (atxr) [inline install.sh:46]: This will probably conflict with #67 :thinking:

**Review by atxr — COMMENTED** [review]
I read the script and it looks very good at first sight.
I have several problems with VM currently so I cannot try to run it right now, but I will do it soon on another computer.

File: install.sh (hunk 1 above)
Comment (amtoine) [reply install.sh:46]: ohh yes it does :wink:
i'll fix them right now :+1:

Comment (amtoine) [reply install.sh]: should be good in 4d1a98b :relieved:

**Comment by amtoine** [comment]
> I read the script and it looks very good at first sight.
>
noice :yum: 

> I have several problems with VM currently so I cannot try to run it right now, but I will do it soon on another computer.
>
no worries, let's take some time to test this :wink:

**Review by atxr — APPROVED** [review]
Awesome work! That's very clean!
Would you like me to test? I mean if you tested on a fresh VM that sounds good for me!

**Comment by amtoine** [comment]
> Awesome work! That's very clean!
>
:yum: 

> Would you like me to test? I mean if you tested on a fresh VM that sounds good for me!
>
here's what we can do:
- try to test this is a VM, possibly by removing the big dependencies such as `goat-paru-git` and `goat-wallpapers-git`, during the afternoon
- if you do not have the time and do not post anything here by, let's say, 6 PM, i'll merge
- otherwise, you might be able to catch a few more things to change before merging :relieved: 

sounds good? :smirk:

**Comment by amtoine** [comment]
it's been 24+h since yesterday so, ready or not, here it comes :muscle:

**Pull request merged**